MSOE.MediaComplete/MainWindow.xaml.cs

```diff
@@ -501,20 +501,25 @@ private void PlaylistTree_SelectionChanged(object sender, RoutedPropertyChangedE
         /// </summary>
         /// <param name="sender"></param>
         /// <param name="e"></param>
-        private void FolderTree_OnSelectionChanged(object sender, EventArgs e)
+        private void FolderTree_OnMouseUp(object sender, EventArgs e)
         {
             if (!IsLoaded)
                 return;
 
-            FormCheck();
-            if (FolderTree.SelectedItems != null)
+            FormCheck(); 
+            var newItems = new List<FolderTreeViewItem>();
+            newItems.AddRange(FolderTree.SelectedItems.Cast<FolderTreeViewItem>());
+            if (FolderTree.SelectedItems != null && FolderTree.SelectedItems.Count >= 0 && !_oldSelectedItems.SequenceEqual(newItems))
             {
+                _oldSelectedItems = newItems;
                 RootLibraryFolderItem.IsSelected = FolderTree.SelectedItems.Count == 0;
                 Songs.View.Refresh();
             }
             ClearDetailPane();
         }
 
+
+        private List<FolderTreeViewItem> _oldSelectedItems = new List<FolderTreeViewItem>();
```

Comment:
I see where you set this. but what is it used for? It looks like all the new code outside the if-block in FolderTree_OnMouseUp is unnecessary.